app/frontend/components/pages/receiveAda/addressItem.tsx

```diff
@@ -1,125 +1,127 @@
-import {h, Component} from 'preact'
-import {connect} from '../../../helpers/connect'
-import actions from '../../../actions'
+import {Fragment, h} from 'preact'
 import {ADALITE_CONFIG} from '../../../config'
 import CopyOnClick from '../../common/copyOnClick'
+import AddressVerification from '../../common/addressVerification'
+import * as QRious from '../../../libs/qrious'
+
+type ViewOnProps = {
+  name: string
+  url: string
+  inline?: boolean
+}
+
+const ViewOn = ({name, url, inline}: ViewOnProps): h.JSX.Element =>
+  inline ? (
+    <a className="address-link" href={url} target="_blank" rel="noopener">
+      {name}
+    </a>
+  ) : (
+    <Fragment>
+      View on{' '}
+      <a className="address-link" href={url} target="_blank" rel="noopener">
+        {name}
+      </a>
+    </Fragment>
+  )
 
 interface Props {
-  openAddressDetail: any
   address: string
   bip32path: string
+  isExpanded: boolean
+  expand: () => void
 }
 
-class AddressItem extends Component<Props, {onSmallDevice: boolean}> {
-  constructor(props) {
-    super(props)
-    this.updateDimensions = this.updateDimensions.bind(this)
-    this.openAddressDetail = this.openAddressDetail.bind(this)
-    this.state = {onSmallDevice: false}
-  }
+const AddressItem = ({address, bip32path, isExpanded, expand}: Props): h.JSX.Element => {
+  const explorerLinksInline = (
+    <div>
+      {ADALITE_CONFIG.ADALITE_CARDANO_VERSION === 'byron' && (
```

Comment:
I won't pester you with refactoring "ADALITE_CARDANO_VERSION === 'byron'" into an enum right now, I'm adding this into the backlog though